R/add_resource.R

```diff
@@ -9,9 +9,17 @@
 #' `.`, `-` and `_`.
 #'
 #' @inheritParams read_resource
-#' @param df A data frame.
-#' @param schema List object describing a Table Schema for that data frame.
-#'   If not provided, one will be created (using [create_schema()]).
+#' @param data Data to attach, either a data frame or path(s) to CSV file(s).
+#'   - Data frame: attached to the resource as `data` and written to a CSV file
+#'     when using [write_package()].
+#'   - Path(s) to CSV file(s): added to the resource as `path`.
```

Comment:
I would add the fact that paths should be passed as a vector. Or do you want to support lists as well? I think a vector with paths is sufficient. But, it should be mentioned in documentation.